patches/server/1054-Added-EntityDamageDoorEvent-and-implemented-into-Ser.patch

```diff
@@ -0,0 +1,44 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: CMarcoo <cmarco.org@gmail.com>
+Date: Thu, 14 Mar 2024 17:04:43 +0100
+Subject: [PATCH] Added EntityDamageDoorEvent and implemented into Server.
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java b/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
+index a85885ee51df585fa11ae9f8fcd67ff2a71c5a18..0d2de8e8747795393631f851cb941b268aa0a4b0 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
++++ b/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
+@@ -1,10 +1,12 @@
+ package net.minecraft.world.entity.ai.goal;
+ 
+ import java.util.function.Predicate;
++import io.papermc.paper.event.entity.EntityDamageDoorEvent;
+ import net.minecraft.world.Difficulty;
+ import net.minecraft.world.entity.Mob;
+ import net.minecraft.world.level.GameRules;
+ import net.minecraft.world.level.block.Block;
++import org.bukkit.craftbukkit.block.impl.CraftDoor;
+ 
+ public class BreakDoorGoal extends DoorInteractGoal {
+ 
+@@ -55,6 +57,20 @@ public class BreakDoorGoal extends DoorInteractGoal {
+     @Override
+     public void tick() {
+         super.tick();
++
++        // Paper start
++        EntityDamageDoorEvent event = new EntityDamageDoorEvent(this.mob.getBukkitEntity(), new CraftDoor(this.mob.level().getBlockState(this.doorPos)), this.breakTime, this.doorBreakTime);
```

Comment:
Creating a CraftDoor like this is pretty meh. paper offers a cached nms.BlockState#createCraftBlockData that we can just yield back under the BlockData type.

The BreakDoorEvent does not offer the block data pre-casted to the Door interface, I am unsure if we want to do this here.

Reply:
Hello, I've been fixing all other comments, however I never heard of this.

How can I access this method?

Reply:
`this.mob.level().getBlockState(this.doorPos).createCraftBlockData()`

in this case. It yields CraftBlockData, not door, but as mentioned previously, I am unsure if we want to force the Door type into the API there anyway, it might be better to just have `BlockData getDamagedDoor()`

Reply:
I changed to this. I'll wait for other's opinion on it.